### PR TITLE
SUS-2119: Semantic MediaWiki - only spawn UpdateJobs for incoming properties on article delete

### PR DESCRIPTION
**Pull request by mszabo-wikia**
SMW PR: https://github.com/Wikia/SemanticMediaWiki/pull/5
Discussion: https://github.com/SemanticMediaWiki/SemanticMediaWiki/issues/2453

JIRA: https://wikia-inc.atlassian.net/browse/SUS-2119

### CODE REVIEW

File: extensions/SemanticMediaWiki/src/MediaWiki/Hooks/ArticleDelete.php [hunk 1]
```diff
@@ -19,74 +23,78 @@
...
 	 */
 	private $wikiPage = null;
 
 	/**
 	 * @since  2.0
 	 *
-	 * @param Wikipage $wikiPage
+	 * @param WikiPage $wikiPage
 	 */
 	public function __construct( &$wikiPage, &$user, &$reason, &$error ) {
```
Comment (WikiaTech) [inline extensions/SemanticMediaWiki/src/MediaWiki/Hooks/ArticleDelete.php:35]: MAJOR Remove the unused function parameter "$user". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
MAJOR Remove the unused function parameter "$reason". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
MAJOR Remove the unused function parameter "$error". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)

File: extensions/SemanticMediaWiki/src/MediaWiki/Hooks/ArticleDelete.php [hunk 2]
```diff
@@ -19,74 +23,78 @@
...
+		$semanticData = new SemanticData( $subject );
 
-			$parameters['semanticData'] = $semanticDataSerializer->serialize(
-				$store->getSemanticData( $subject )
-			);
+		$properties = $store->getInProperties( $subject );
 
-			$jobFactory->newUpdateDispatcherJob( $title, $parameters )->insert();
+		foreach ( $properties as $property ) {
+			// Avoid doing $propertySubjects = $store->getPropertySubjects( $property, $subject );
```
Comment (WikiaTech) [inline extensions/SemanticMediaWiki/src/MediaWiki/Hooks/ArticleDelete.php:74]: MAJOR Remove this commented out code. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS125)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 5 issues
* ![MAJOR][MAJOR] 5 major

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [UpdateDispatcherJob.php#L183](https://github.com/Wikia/app/blob/fcaf738f5a0b5991753069fe8806eb09f35c3f8d/extensions/SemanticMediaWiki/src/MediaWiki/Jobs/UpdateDispatcherJob.php#L183): Remove this commented out code. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS125)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: extensions/SemanticMediaWiki/src/MediaWiki/Hooks/ArticleDelete.php (hunk 2 above)
Comment (mszabo-wikia) [reply extensions/SemanticMediaWiki/src/MediaWiki/Hooks/ArticleDelete.php:74]: invalid

File: extensions/SemanticMediaWiki/src/MediaWiki/Hooks/ArticleDelete.php (hunk 1 above)
Comment (mszabo-wikia) [reply extensions/SemanticMediaWiki/src/MediaWiki/Hooks/ArticleDelete.php:35]: this is a hook handler, and it's in SMW

**Review by macbre — APPROVED** [review]

**Pull request merged**